internal/manifest/oci_index.go

```diff
@@ -102,13 +115,26 @@ func (index *OCI1IndexPublic) editInstances(editInstances []ListEdit) error {
 				return fmt.Errorf("update %d of %d passed to OCI1Index.UpdateInstances had no media type (was %q)", i+1, len(editInstances), index.Manifests[i].MediaType)
 			}
 			index.Manifests[targetIndex].MediaType = editInstance.UpdateMediaType
+			for k, v := range editInstance.UpdateAnnotations {
+				index.Manifests[targetIndex].Annotations[k] = v
+			}
```

Comment:
`x/exp/maps.Clone` please, then we don’t have to even think about correctness of the implementation.

---

The above would, AFAICS, preserve any old annotations in the descriptor, i.e. it would not allow removing an annotation. Using `Clone` + overwriting `.Annotations` is clearly an overwrite.